exporter/awskinesisexporter/internal/batch/batch.go

```diff
@@ -98,18 +108,45 @@ func (b *Batch) AddRecord(raw []byte, key string) error {
 		return ErrPartitionKeyLength
 	}
 
-	if l := len(record); l == 0 || l > b.maxRecordSize {
+	if l := len(compressed); l == 0 || l > b.maxRecordSize {
 		return ErrRecordLength
 	}
 
-	b.records = append(b.records, &kinesis.PutRecordsRequestEntry{Data: record, PartitionKey: aws.String(key)})
+	if b.useAggregation && b.aggregator.IsRecordAggregative(compressed, key) {
+		if b.aggregator.IsBatchFull(compressed, key) {
+			record, err := b.aggregator.Drain()
+			if err != nil {
+				log.Fatal(err)
+			}
+			if record != nil {
+				b.records = append(b.records, record)
+			}
+		} else {
+			b.aggregator.Put(compressed, key)
+		}
+	} else {
+		b.records = append(b.records, &kinesis.PutRecordsRequestEntry{Data: compressed, PartitionKey: aws.String(key)})
+	}
 	return nil
 }
 
+func (b *Batch) Close() {
+	if !b.useAggregation {
+		return
+	}
+	record, err := b.aggregator.Drain()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
The collector doesn't allow for calls to `fatal` loggers, since it would shutdown the entire process.